components/script/dom/window.rs

```diff
@@ -2001,7 +1998,8 @@ impl Window {
         }
 
         let document = self.Document();
-        let font_face_set = document.Fonts(can_gc);
+        document.update_fonts_state();
```

Comment:
This is incorrect-we should replace the fulfill_ready_promise_if_needed with this call instead, otherwise we are marking the promise as fulfilled unconditionally.